src/Compilers/Core/Rebuild/CompilationOptionsReader.cs

```diff
@@ -136,15 +136,12 @@ public ImmutableArray<MetadataReferenceInfo> GetMetadataReferences()
             ? OutputKind.ConsoleApplication
             : OutputKind.DynamicallyLinkedLibrary;
 
-        public string? GetMainTypeName() => GetMainMethodInfo() is { } tuple
-            ? tuple.MainTypeName
-            : null;
+        // Here we only want to give the caller the main type name if the main method is named "Main" per convention.
+        // If the main method has another name, we have to assume that specifying a main type name won't work.
+        // For example, if the compilation uses top-level statements.
```

Comment:
Any reason you didn't make this an XML doc?

Reply:
I moved this to an inline comment for now. I am fine with it being either inline or XML doc, but it seems like it would be more appropriate on the "root" `GetMethodMainInfo` if it were moved to XML doc.

It's not currently clear to me that CompilationOptionsReader is going to be part of the public API in the long term. For example, perhaps we will go back to accepting the PEReader+MetadataReader combo directly out of the feeling that we don't need to provide the individual facilities of the CompilationOptionsReader to library consumers.